core/include/vecmem/memory/device_atomic_ref.hpp

```diff
@@ -213,4 +213,17 @@ class device_atomic_ref {
 // Include the implementation.
 #include "vecmem/memory/impl/device_atomic_ref.ipp"
 
+#if __cpp_concepts >= 201907L
+#include "vecmem/concepts/atomic_ref.hpp"
+static_assert(
+    vecmem::concepts::atomic_ref<vecmem::device_atomic_ref<uint32_t> >,
+    "Atomic reference on uint32_t is incompletely defined.");
+static_assert(
+    vecmem::concepts::atomic_ref<vecmem::device_atomic_ref<uint64_t> >,
+    "Atomic reference on uint64_t is incompletely defined.");
+static_assert(
+    vecmem::concepts::atomic_ref<vecmem::device_atomic_ref<std::size_t> >,
+    "Atomic reference on std::size_t is incompletely defined.");
+#endif
```

Comment:
:thinking: Is this the place to do this check? I'm honestly not sure...

Reply:
I think it's fine here. You get better diagnostics if you check for concepts in a template argument, but we don't really use these atomic references for that (yet?). Still, I think checking for these is very important if we're going to sometimes use `sycl::atomic_ref`, etc.